### ISSUE TITLE
createPaymentMethod only works for credit cards (Android)

### ISSUE THREAD

**Issue by ghitier** [stripe/stripe-react-native#445]
**Describe the bug**
On android `createPaymentMethod` only works with credit cards

**To Reproduce**

Call:
```javascript
const { paymentMethod, error } =  await createPaymentMethod({
  type: 'SepaDebit',
    iban,
    billingDetails: {
      email: 'john.doe@example.com',
      name: 'John DOE',
    },
})
```

You will get an error: **`Card details not complete`**

**Expected behavior**
It should complete the PM creation successfully

**Screenshots**
~~If applicable, add screenshots to help explain your problem.~~

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version: Big Sur

**Smartphone (please complete the following information):**
 - Device: OnePlus 5T
 - OS: Android
 - Version: 10

**Additional context**
After looking at the source code it seems to me that the `createPaymentMethod` (in *`StripeSdkModule.kt`*), makes direct  use of the stripe SDKs `PaymentMethodCreateParams.create(...)` (providing the details from the CardField); Compared to the confirm and setup intents which both use factories from `PaymentMethodCreateParamsFactory`.

**Comment by thorsten-stripe** [comment]
For SEPA DD please follow the docs here:
- https://stripe.com/docs/payments/sepa-debit/accept-a-payment?platform=react-native
- https://stripe.com/docs/payments/sepa-debit/set-up-payment?platform=react-native

**Comment by ghitier** [comment]
Hi @thorsten-stripe thank you for your suggestion to use setup and confirm directly. It seems it would indeed help to simplify my implementation. To clarify, this was my current one: Create a PM > Setup or Confirm with this PM (id).

**Nonetheless I think this is a bug.** Here's a few reasons.

1. The Typescript definitions for this  library:

```typescript
createPaymentMethod: (data: Params, options?: Options) => Promise<CreatePaymentMethodResult>;

confirmPayment: (paymentIntentClientSecret: string, data: Params, options?: Options) => Promise<ConfirmPaymentMethodResult>

confirmSetupIntent: (paymentIntentClientSecret: string, data: Params, options?: Options) => Promise<ConfirmSetupIntentResult>
```

as can be seen they all have the same `data: Param`, I would think that I could use them with the same arguments.

2. Creating works as expected on iOS. Furthermore nothing seems to prevent it in the native android SDK.

3. According to the Stripe JS documentation:
```
Use stripe.createPaymentMethod to convert payment information collected by elements into a PaymentMethod object that you safely pass to your server to use in an API call.

NOTE: In most integrations, you will not need to use this method. Instead, use methods like stripe.confirmCardPayment, which will automatically create a PaymentMethod when you confirm a PaymentIntent.
```

My understanding is that the approach of using createPaymentMethod first instead of confirmPaymentIntent/confirmSetupIntent, is  therefore valid but not suggested (being somewhat more low-level)

**To summarise: although not suggested, In my opinion createPaymentMethod should still support the same arguments as confirm and setup, unless stated otherwise**.

---

Also here is my take on fixing this: https://github.com/Freshmile/stripe-react-native/commit/3906111831234826f825ce520c126ef9f4e85fb5 *(only for credit card and sepa currently)*
